src/app/datapoints-graph/charts/y-axis.service.ts

```diff
@@ -22,42 +22,101 @@ export class YAxisService {
       { position: DatapointAxisType; offset: number }
     > = this.getYAxisPlacement(datapointsWithValues);
 
-    return datapointsWithValues.map((dp) => ({
-      type: 'value',
-      animation: true,
-      axisLine: {
-        show: true,
-        lineStyle: {
-          color: dp.color,
+    const matchingDpSet = new Set<DatapointWithValues>();
+    const firstOccurrence = new Set<DatapointWithValues>();
+
+    return datapointsWithValues.map((dp, index) => {
+      const matchingDpRange = datapointsWithValues.some(
+        (dp2, index2) =>
+          dp2.min === dp.min && dp2.max === dp.max && index2 < index
+      );
+
+      const anyMatchingDp = datapointsWithValues.some(
+        (dp2, index2) =>
+          dp2.min === dp.min && dp2.max === dp.max && index2 !== index
+      );
+
+      if (
+        anyMatchingDp &&
+        !matchingDpRange &&
+        YAxisOptions.mergeMatchingDatapoints &&
+        !firstOccurrence.has(dp)
+      ) {
+        firstOccurrence.add(dp);
+      }
+
+      if (firstOccurrence.has(dp)) {
+        datapointsWithValues.forEach((dp2) => {
+          if (dp2.min === dp.min && dp2.max === dp.max) {
```

Comment:
I guess this will not work. As if both are undefined it will still merge them, which does not make sense. I guess we need to check for undefined here as well and if undefined don't merge.

Look at this graph which shows only one axis, but the axis are very different (altitude is between 2000 and 2200) but only the axis for BatteryLevel is shown.
![image](https://github.com/user-attachments/assets/fa3c289a-0901-4111-9328-00b0a58edd5d)